src/System.Private.ServiceModel/src/System/ServiceModel/Security/SecurityUtils.cs

```diff
@@ -455,6 +456,168 @@ internal static string GenerateId()
             return SecurityUniqueId.Create().Value;
         }
 
+        internal static ReadOnlyCollection<IAuthorizationPolicy> CreatePrincipalNameAuthorizationPolicies(string principalName)
+        {
+            if (principalName == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("principalName");
+
+            Claim identityClaim;
+            Claim primaryPrincipal;
+            if (principalName.Contains("@") || principalName.Contains(@"\"))
+            {
+                identityClaim = new Claim(ClaimTypes.Upn, principalName, Rights.Identity);
+#if FEATURE_CORECLR
+                primaryPrincipal = Claim.CreateUpnClaim(principalName);
+#else 
+                throw ExceptionHelper.PlatformNotSupported("UPN claim not supported on CoreCLR"); 
+#endif // FEATURE_CORECLR
```

Comment:
Oops, this message needs to change. However, FEATURE_CORECLR is correct because of this:
https://github.com/dotnet/wcf/blob/master/src/System.Private.ServiceModel/src/System/IdentityModel/Claims/Claim.cs#L148

 so we should be consistent here.